lib/datadog/appsec/contrib/rack/request_middleware.rb

```diff
@@ -26,7 +26,24 @@ def initialize(app, opt = {})
           def call(env)
             return @app.call(env) unless Datadog::AppSec.enabled?
 
-            Datadog::Core::Remote.active_remote.barrier(:once) unless Datadog::Core::Remote.active_remote.nil?
+            unless Datadog::Core::Remote.active_remote.nil?
+              barrier = nil
+
+              t = Datadog::Core::Utils::Time.measure do
+                barrier = Datadog::Core::Remote.active_remote.barrier(:once)
+              end
+
+              if barrier != :pass && (span = active_span) && !span.has_tag?('_dd.rc.boot.time')
+                span.set_tag('_dd.rc.boot.time', t)
+
+                if barrier == :timeout
+                  span.set_tag('_dd.rc.boot.timeout', true)
+                else
+                  # TODO: 'ready' should evolve into ensuring RC received a proper reply
+                  span.set_tag('_dd.rc.boot.ready', true)
+                end
+              end
+            end
```

Comment:
Is it possible to encapsulate/review this logic, so it's compartmentalized and shard with other occurrence on `lib/datadog/tracing/contrib/rack/middlewares.rb`?

My motivation for asking this is because it's read like an unrelated `unless/end` block in the middle of this `call` method. I wouldn't mind it if it was hidden behind an abstraction.

Reply:
Indeed. I initially set out to do that, especially given that the tracing side does exactly the same thing except that on the tracing side it's split in two because of the span creation.

I'll try something.